src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
     });
 }
 
+fn inlay_hints_from_editor_code_lens(
+    meta: EditorMeta,
+    mut lenses: Vec<CodeLens>,
+    ctx: &mut Context,
+) {
+    let document = match ctx.documents.get(&meta.buffile) {
+        Some(document) => document,
+        None => {
+            return;
+        }
+    };
+    lenses.sort_by_key(|lens| lens.range.start);
+
+    let inlay_hints_from_lenses = lenses
+        .iter()
+        .map(|lens| {
+            let label = lens.command.as_ref().map_or("", |v| &v.title);
+            if label.trim().len() > 0 {
+                let position =
+                    lsp_position_to_kakoune(&lens.range.start, &document.text, ctx.offset_encoding);
+                let line = position.line;
+                let col = position.column;
+                editor_quote(&format!(
+                    "{line}.{col}+0|{{InlayHint}}[\u{1F50E} {}] {{InlayHint}}",
```

Comment:
I like the emoji :)
if we added a test, we would probably see it fail in macOS CI because their glibc uses (?) an old version of Unicode.
This means macOS users might see glitches when we use emoji.
Our workaround is 73c2d1ab (Work around inconsistent wcwidth by using ASCII instead of emoji, 2022-01-21).
It should be fairly easy to extend this to this emoji (look for `lsp_emoji_lightbulb_ok`). Although we should probably document the workaround better.

Reply:
Ok, I will try to incorporate the workaround.

Reply:
> As emoji's become more important generally, it would be useful to always know if we can sent emoji's or not.

I agree but instead of adding an request parameter (which won't ever change) we should just move the awkward computation of wcwidth (see `*lsp-scratch*`) into the controller code. From Rust we can easily call `wcwidth` and then just set the options during initialization. Note that it's easy to simulate the bad wcwidth with `LC_ALL=C` (though such a non-UTF-8 locale might cause other issue, so it may not be worth supporting).

We can always use the emoji for now and leave the wcwidth-probing to a follow up (lsp-inlay-code-lenses are an optional feature after all)